packages/client/hmi-client/src/components/workflow/ops/compare-datasets/tera-compare-datasets-drilldown.vue

```diff
@@ -262,60 +293,99 @@ function findDuplicates(strings: string[]): string[] {
-	if (timepointIndex === undefined) return;
+	if (timepointIndex === undefined || timepointIndex === -1) return;
+
+	// Find dataset index of the selected dataset
+	const selectedIndex = datasets.value.findIndex((dataset) => dataset.id === knobs.value.selectedDataset);
+	if (selectedIndex === -1) return;
 
-	// Go through every column header
+	const { selectedPlotType } = knobs.value;
+
+	// Go through every common header (column loop)
 	commonHeaderNames.value?.forEach((headerName) => {
 		if (headerName === timepointHeaderName.value) return;
 
 		const headerIndex = transposedRawContents[0]?.headers?.indexOf(headerName);
 		if (!headerIndex) return;
 
-		const values: any = [];
-		// Find dataset index of the selected dataset
-		const selectedIndex = datasets.value.findIndex((dataset) => dataset.id === knobs.value.selectedDataset);
-		const referenceColumn = transposedRawContents[selectedIndex].csv[headerIndex]; // aka the baseline dataset
-
-		transposedRawContents.forEach((content, datasetIndex) => {
-			const timepoints = content.csv[timepointIndex];
-			timepoints.forEach((timepoint: number, rowIndex: number) => {
-				const referencePoint = parseFloat(referenceColumn[rowIndex]);
-				const currentPoint = parseFloat(content.csv[headerIndex][rowIndex]);
-
-				const absoluteDifference = referencePoint - currentPoint;
-				const percentChange = (absoluteDifference / referencePoint) * 100;
-
-				values.push({
-					timepoint,
-					value: knobs.value.selectedPlotValue === PlotValue.VALUE ? absoluteDifference : percentChange,
-					name: `${headerName}_${datasets.value[datasetIndex].name}`
-				});
+		const data: any = [];
+		const variableNames: string[] = [];
+		const referenceColumn = transposedRawContents[selectedIndex].csv[headerIndex]; // aka the column of baseline dataset we are subtracting from
+
+		transposedRawContents.forEach(({ csv }, datasetIndex) => {
+			const timepoints = csv[timepointIndex];
+			const columnToSubtract = csv[headerIndex];
+
+			const name = `${headerName}_${datasets.value[datasetIndex].name}`;
+			variableNames.push(name);
+
+			referenceColumn.forEach((referencePoint: number, index: number) => {
+				let value = 0;
+				if (selectedPlotType === PlotValue.DIFFERENCE) {
+					value = referencePoint - columnToSubtract[index]; // difference
+				} else if (selectedPlotType === PlotValue.PERCENTAGE) {
+					value = ((referencePoint - columnToSubtract[index]) / referencePoint) * 100; // percentage
+				} else if (selectedPlotType === PlotValue.VALUE) {
+					value = parseFloat(columnToSubtract[index]); // trajectory
+				}
+				if (data[index] === undefined) {
+					data.push({
+						[name]: value,
+						timepoint: parseFloat(timepoints[index])
+					});
+				} else {
+					data[index][name] = value;
+				}
 			});
 		});
-
-		compareCharts.value.push(createDatasetCompareChart(values, headerName));
+		compareCharts.value.push(
```

Comment:
It looks like we can just re use `useVariableCharts` in `useCharts.ts` instead of creating a chart manually here. You may need to modify some logic inside `useVariableCharts` on how it creates the chart options to provide correct variable names though.